core/workflow/loadDPL.go

```diff
@@ -57,5 +62,95 @@ func LoadDPL(tasks []*task.Class) (workflow Role, err error) {
 	}
 
 	workflow = root
+
+	// FIXME: either get rid of err or add handling of errors
 	return workflow, nil
 }
+
+// Aux struct to fulfil export requirement by yaml.Marshal
+type auxAggregatorRole struct {
+	RoleBase   roleBase
+	Aggregator aggregator
+}
+
+func RoleToYAML(input Role) ([]byte, error) {
```

Comment:
Trying to use `Role` as a receiver for `MarshalYAML` yields an `Invalid receiver type 'Role'`. I decoupled the `RoleToYAML` function to pass `auxRole` as the receiver for Marshller. It does seem a bit hacky. Let me know if there's a better way to approach this.

Reply:
Of course, note that I never suggested `(Role) MarshalYAML` as I'm aware that this is impossible. But I assert that `(* aggregatorRole) MarshalYAML` is possible, and should be enough without additional, non-local auxiliary types (also related to the discussion on using `map[string]interface{}` as auxiliary format.

Reply:
I find these two `RoleToYAML` functions a bit confusing. This one accepts a `Role` as input, which internally is presumably an `*aggregatorRole`. Instead of creating a new `aggregatorRole` here only to call `yaml.Marshal`, you can type-assert your `input` to `*aggregatorRole` instead.

Reply:
This is fantastic! Now the `RoleToYAML` function exists solely as a wrapper for `yaml.Marshal` which allows us to avoid exporting `aggregatorRole`.